packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Providers/ModelFactoryProvider.cs

```diff
@@ -271,7 +265,7 @@ private static bool TryBuildMethodArgumentsForOverload(
                 return false;
             }
 
-            var currentParameters = currentMethod.Parameters.ToHashSet();
+            var currentParameters = currentMethod.Parameters.ToHashSet(ParameterProvider.EqualityByNameAndType);
```

Comment:
since in this context, we are trying to compare the parameters by their names and types, we used a comparer here to do that, and its default behavior in a dictionary is by reference equality.